zipline/data/data_portal.py

```diff
@@ -843,8 +857,10 @@ def _get_minute_window_for_asset(self, asset, field, minutes_for_window):
             return self._get_minute_window_for_future(asset, field,
                                                       minutes_for_window)
         else:
-            return self._get_minute_window_for_equity(asset, field,
-                                                      minutes_for_window)
+            # TODO: Make caller accept assets.
```

Comment:
No, I mean that `_get_history_minute_window` can and should pass an iterable of assets to `_get_minute_window_for_asset(s)` instead of calling multiple times.

With that change the start and end locations would only need to be calculated once per bar, the multiplication for the thousands place can be done only once, etc.